First draft documentation for GPU config in Kayobe

oneswig opened the pull request:
Based on Isaacs playbook in kayobe-ops and other work by
Will and the SIB team.

Needs example config for PCI passthrough in Nova.

oneswig committed:
First draft documentation for GPU config in Kayobe

Based on Isaacs playbook in kayobe-ops and other work by
Will and the SIB team.

priteau commented on source/gpus_in_openstack.rst:
```diff
@@ -0,0 +1,217 @@
+Support for GPUs in OpenStack
+=============================
+
+This guide is has been developed for Nvidia GPUs and CentOS 8.
```
s/is /

Nit: s/Nvidia/NVIDIA/

priteau commented on source/gpus_in_openstack.rst:
```diff
@@ -0,0 +1,217 @@
+* https://www.jimmdenton.com/gpu-offloading-openstack/
+* https://docs.openstack.org/nova/latest/admin/pci-passthrough.html
+* https://docs.openstack.org/nova/latest/admin/virtual-gpu.html (vGPU only)
+* Telsa models in OpenStack: https://egallen.com/openstack-nvidia-tesla-gpu-passthrough/
```
Tesla

oneswig committed:
Add example config for Nova compute service

oneswig committed:
Feedback from Pierre

bbezak committed:
aditional GPU Passhtrough configuration bits:
- nova-scheduler, nova-api configuration
- flavor and instance creation
- iommu testing in nova_libvirt

oneswig commented on source/gpus_in_openstack.rst:
```diff
@@ -0,0 +1,313 @@
+
+.. code-block:: text
+
+   openstack flavor set m1.medium --property "pci_passthrough:alias"="gpu-p100:2"
```
Can we make a reference here to the project-config repo, eg https://github.com/stackhpc/openstack-admin-guide/blob/master/source/vars.rst#L29

oneswig reviewed (commented):
Would be great to link in the project-config repo for flavor configuration

bbezak committed:
added flavor configuration for GPU in project_config
changed subsubsection characters to more conventional ones

oneswig reviewed (commented):
LGTM

Wasaac reviewed (approved):
LGTM

priteau commented on source/gpus_in_openstack.rst:
```diff
@@ -0,0 +1,341 @@
+pci.alias also needs to be configured on the controller. 
+This configuration should match the configuration found on the compute nodes.
+Add it to Kolla-Ansible configuration file:
+``etc/kayobe/kolla/config/nova-api/nova.conf``, for instance:
```
This isn't the correct path, it should be:

    etc/kayobe/kolla/config/nova/nova-api.conf

priteau commented on source/gpus_in_openstack.rst:
```diff
@@ -0,0 +1,341 @@
+
+.. code-block:: text
+
+   kayobe overcloud service reconfigure -kt nova --kolla-skip-tags common --skip-precheck
```
s/-kt/--kolla-tags/
s/--skip-precheck/--skip-prechecks/

bbezak committed:
nova-api configure path fix, --skip-prechecks
